### PR TITLE
Add federal income tax as a Montana itemized deduction

### LINKED ISSUE #3427: Add federal income tax as a Montana itemized deduction
The [2021 Montana income tax instructions](https://mtrevenue.gov/wp-content/uploads/mdocs/form%202%202021%20instructions.pdf#page=32) make it clear that up to $5,000 (or $10,000 for joint filers) of federal income tax liability (that is, the `income_tax` variable) can be a Montana itemized deduction.   The fact that this feature of the Montana income tax is not yet included in Policyengine-US means that there are thousands of differences with TAXSIM35 even in the simple `e21` sample that includes no expenses that are itemizable on a federal return.

### PR DESCRIPTION
**Pull request by Shinuing**
Fixes #3427

### CODE REVIEW

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/itemized/federal_income_tax_limit.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  label: Montana federal income tax deduction limit
+  period: year
+  unit: currency-USD
+  breakdown:
+    - filing_status
+  reference:
+    - title: MT Code § 15-30-2131 (2021) (1)(b)
+      href: https://law.justia.com/codes/montana/2021/title-15/chapter-30/part-21/section-15-30-2131/
+    - title: 2021 Montana Form 2 Indicadual Income Tax Instructions
+      href: https://mtrevenue.gov/wp-content/uploads/mdocs/form%202%202021%20instructions.pdf#page=32
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/itemized/federal_income_tax_limit.yaml:22]: could you add the 2022 reference to see if we have inflation adjustments

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/itemized/federal_income_tax_limit.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/itemized/federal_income_tax_limit.yaml:1]: make a federal_income_tax folder and rename this file to cap.yaml

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/itemized/federal_income_tax_limit.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+description: Montana allows a federal income tax deduction of up to this amount based on filing status.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/itemized/federal_income_tax_limit.yaml:1]:
```suggestion
description: Montana caps the federal income tax deduction to this amount, based on filing status.
```

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/itemized/federal_income_tax_limit.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+HEAD_OF_HOUSEHOLD:
+  2021-01-01: 5_000
+WIDOW:
+  2021-01-01: 5_000
+SINGLE:
+  2021-01-01: 5_000
+SEPARATE:
+  2021-01-01: 5_000
+metadata:
+  label: Montana federal income tax deduction limit
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/itemized/federal_income_tax_limit.yaml:13]:
```suggestion
  label: Montana federal income tax deduction cap
```

File: changelog_entry.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Add federal income tax as a Montana itemized deduction.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Montana federal income tax deduction.
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/gov/states/mt/tax/income/deductions/itemized/mt_federal_income_tax_deduction.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    defined_for = StateCode.MT
+
+    def formula(tax_unit, period, parameters):
+        filing_status = tax_unit("filing_status", period)
+        p = parameters(
+            period
+        ).gov.states.mt.tax.income.deductions.itemized.federal_income_tax
+        federal_income_tax_limit = p.cap[filing_status]
+
+        federal_income_tax = tax_unit("income_tax", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mt/tax/income/deductions/itemized/mt_federal_income_tax_deduction.py:23]: As the tax form requests `Federal income tax withheld` plus `Federal estimated tax payments` I don't think it includes negative values from refundable credits.
```suggestion
        federal_income_tax = tax_unit("income_tax_before_refundable_credits", period)
```

**Review by MaxGhenis — APPROVED** [review]
good to merge after this

**Pull request merged**